sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/implementation/WebExceptionRetryPolicyTest.java

```diff
@@ -0,0 +1,318 @@
+        BridgeInternal.setSubStatusCode(cosmosException, HttpConstants.SubStatusCodes.GATEWAY_ENDPOINT_READ_TIMEOUT);
+
+        RxDocumentServiceRequest dsr;
+        Mono<ShouldRetryResult> shouldRetry;
+
+        //Data Plane Write - Should not retry
+        dsr = RxDocumentServiceRequest.createFromName(mockDiagnosticsClientContext(),
+            OperationType.Create, "/dbs/db/colls/col/docs/doc", ResourceType.Document);
+
+        WebExceptionRetryPolicy webExceptionRetryPolicy = new WebExceptionRetryPolicy(new RetryContext());
+        webExceptionRetryPolicy.onBeforeSendRequest(dsr);
+        shouldRetry = webExceptionRetryPolicy.shouldRetry(cosmosException);
+
+        validateSuccess(shouldRetry, ShouldRetryValidator.builder()
+            .nullException()
+            .shouldRetry(false)
+            .build());
+
+        //Metadata Write - Should not Retry
+        dsr = RxDocumentServiceRequest.createFromName(mockDiagnosticsClientContext(),
+            OperationType.Create, "/dbs/db/colls/col/docs/docId", ResourceType.Database);
+
+        webExceptionRetryPolicy = new WebExceptionRetryPolicy(new RetryContext());
+        webExceptionRetryPolicy.onBeforeSendRequest(dsr);
+        shouldRetry = webExceptionRetryPolicy.shouldRetry(cosmosException);
+
+        validateSuccess(shouldRetry, ShouldRetryValidator.builder()
+            .nullException()
+            .shouldRetry(false)
+            .build());
+    }
+
+    @Test(groups = "unit")
+    public void httpNetworkFailureOnAddressRefresh() throws Exception {
+        GlobalEndpointManager endpointManager = Mockito.mock(GlobalEndpointManager.class);
+        Mockito.doReturn(new URI("http://localhost:")).when(endpointManager).resolveServiceEndpoint(Mockito.any(RxDocumentServiceRequest.class));
+        Mockito.doReturn(Mono.empty()).when(endpointManager).refreshLocationAsync(Mockito.eq(null), Mockito.eq(false));
+        Mockito.doReturn(2).when(endpointManager).getPreferredLocationCount();
+
+        RetryContext retryContext = new RetryContext();
+        WebExceptionRetryPolicy webExceptionRetryPolicy = new WebExceptionRetryPolicy(retryContext);
+        Exception exception = ReadTimeoutException.INSTANCE;
+        CosmosException cosmosException = BridgeInternal.createCosmosException(null, HttpConstants.StatusCodes.REQUEST_TIMEOUT, exception);
+        BridgeInternal.setSubStatusCode(cosmosException, HttpConstants.SubStatusCodes.GATEWAY_ENDPOINT_READ_TIMEOUT);
+
+        RxDocumentServiceRequest dsr = RxDocumentServiceRequest.createFromName(mockDiagnosticsClientContext(),
+            OperationType.Read, "/dbs/db/colls/col/docs/", ResourceType.Document);
+        dsr.setAddressRefresh(true, false);
+        dsr.requestContext = new DocumentServiceRequestContext();
+
+        webExceptionRetryPolicy.onBeforeSendRequest(dsr);
+
+        for (int i = 0; i < 10; i++) {
+            Mono<ShouldRetryResult> shouldRetry = webExceptionRetryPolicy.shouldRetry(cosmosException);
```

Comment:
what about the response timeout value validation? 
and also for address refresh, there is no backoff time differences between the retries?

Reply:
Fixed the test case